pkgs/development/libraries/libmcrypt/default.nix

```diff
@@ -18,6 +18,10 @@ stdenv.mkDerivation rec {
       "ac_cv_func_realloc_0_nonnull=yes"
     ];
 
+    env = lib.optionalAttrs stdenv.cc.isClang {
+      NIX_CFLAGS_COMPILE = "-Wno-implicit-function-declaration -Wno-implicit-int";
+    };
```

Comment:
```suggestion
    env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-implicit-function-declaration -Wno-implicit-int";
```

Reply:
What's the advantage of doing it that way? With `optionalAttrs` you don't set `NIX_CFLAGS_COMPILE` to an empty string.

Though I just noticed there's an indentation error there so lemme fix that real quick.

Reply:
Just more short form, I guess empty string is not a problem (or not?). But no problem, up to you)
Out of curiosity:
```sh
$ ag -c "env = lib.optionalAttrs stdenv.cc.isClang" | wc -l
      34
$ ag -c "NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang" | wc -l
      28
```

Reply:
The only difference I can think of is that `optionalAttrs` doesn't invalidate existing Linux builds unnecessarily. I think.